eessi/testsuite/tests/apps/walberla/walberla.py

```diff
@@ -0,0 +1,154 @@
+https://reframe-hpc.readthedocs.io/en/stable/regression_test_api.html#test-variants
+
+"concrete test cases": all test combinations that will actually run:
+- test variants
+- valid system:partition+programming environment combinations
+https://reframe-hpc.readthedocs.io/en/stable/tutorial_deps.html#listing-dependencies
+
+Tests can be filtered by name, tag, programming environment, system, partition, or maintainer,
+see https://reframe-hpc.readthedocs.io/en/stable/manpage.html#test-filtering
+
+Hooks acting on all possible test combinations (before filtering) are called after the 'init' stage.
+Hooks acting on concrete test cases (after filtering) are called after the 'setup' stage.
+
+See also https://reframe-hpc.readthedocs.io/en/stable/pipeline.html
+"""
+
+import reframe as rfm
+# added only to make the linter happy
+from reframe.core.builtins import deferrable, parameter, run_after, run_before, sanity_function, performance_function
+import reframe.utility.sanity as sn
+from eessi.testsuite.constants import COMPUTE_UNITS, DEVICE_TYPES, SCALES
+from eessi.testsuite.eessi_mixin import EESSI_Mixin
+from eessi.testsuite.utils import find_modules
+
+
+def filter_scales_1M():
+    """
+    Filtering function for filtering scales for the waLBerla test
+    returns all scales with at least half a node.
+    """
+    return [
+        k for (k, v) in SCALES.items()
+        if ((v['num_nodes'] <= 2) and (v.get('node_part', 0) != 0)) or (v.get('num_cpus_per_node', 0)
+                                                                        * v.get('num_nodes', 0) > 1)
+    ]
+
+
+@rfm.simple_test
+class EESSI_WALBERLA_BACKWARD_FACING_STEP(rfm.RunOnlyRegressionTest, EESSI_Mixin):
+    """
+    This is the main walberla class for the backward facing step test. The test consists of steps that modify the input
+    file from the precompiled walberla tutorials binary. These are run as pre-run commands and the main test with the
+    executable `05_BackwardFacingStep` is measured for performance.
+    """
+#    ldc_64M = fixture(EESSI_OPENFOAM_base, scope='partition')
+    executable = '05_BackwardFacingStep'
+    executable_opts = ['05_BackwardFacingStep.prm']
+    time_limit = '120m'
+    readonly_files = ['']
+    device_type = parameter([DEVICE_TYPES.CPU])
+    module_name = parameter(find_modules('waLBerla'))
+    valid_systems = ['*']
+    scale = parameter(filter_scales_1M())
+
+    @run_after('init')
+    def set_compute_unit(self):
+        """
+        Set the compute unit to which tasks will be assigned:
+        one task per CPU core for CPU runs.
+        """
+        if self.device_type == DEVICE_TYPES.CPU:
+            self.compute_unit = COMPUTE_UNITS.CPU
+        else:
+            msg = f"No mapping of device type {self.device_type} to a COMPUTE_UNITS was specified in this test"
+            raise NotImplementedError(msg)
+
+    def required_mem_per_node(self):
+        return self.num_tasks_per_node * 1700
```

Comment:
Is there really no offset?

Reply:
Just measured it. First is Rome + foss-2022b, then Rome + foss-2023a, then Genoa + foss-2023a (core count vs max memory usage in the cgroup):

```
8	1485
16	2507
32	4765
64	9394
128	19686
128	25651
	
	
8	884
16	1460
32	2507
64	4595
128	9172
128	10162
	
	
8	1803
24	3013
48	5592
96	10843
192	24009

```

This gives me 179.6x-673, 73.466+192, 121.9+79 in terms of linear equations. I'd say 300 MB per core + 400 MB offset is a very conservative estimate here.